src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -208,8 +208,17 @@ internal override void WritePropertyRefElement(IEdmStructuralProperty property)
         {
             // Key properties without a key alias are represented as strings containing the property name.
             // Key properties with a key alias are represented as objects with one member whose name is the key alias and whose value is a string containing the path to the property.
-            // TODO: It seems the second one is not supported.
-            this.jsonWriter.WriteStringValue(property.Name);
+            if (property is IEdmStructuralPropertyAlias propertyAlias)
+            {
+                this.jsonWriter.WriteStartObject();
+                this.jsonWriter.WritePropertyName(propertyAlias.PropertyAlias);
+                this.jsonWriter.WriteStringValue(string.Join("/", propertyAlias.Path));

```

Comment:
This `join` operation is done in quite a few places. Could we have taken care of it by including a `readonly` property to the `IEdmStructuralPropertyAlias` interface and it's derivations that returns the concatenated thing? We could even apply an optimization there by doing the concatenation only once when the getter is first called and saving the result in a private field